google/resource_sql_database_instance.go

```diff
@@ -442,260 +442,36 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
-						entry.Name = vpp.(string)
-					}
-
-					if vpp, okpp := _entry["value"]; okpp {
-						entry.Value = vpp.(string)
-					}
-
-					settings.IpConfiguration.AuthorizedNetworks = append(
-						settings.IpConfiguration.AuthorizedNetworks, entry)
-				}
-			}
-		}
-	}
-
-	if v, ok := _settings["location_preference"]; ok {
-		_locationPreferenceList := v.([]interface{})
-
-		if len(_locationPreferenceList) == 1 && _locationPreferenceList[0] != nil {
-			settings.LocationPreference = &sqladmin.LocationPreference{}
-			_locationPreference := _locationPreferenceList[0].(map[string]interface{})
-
-			if vp, okp := _locationPreference["follow_gae_application"]; okp {
-				settings.LocationPreference.FollowGaeApplication = vp.(string)
-			}
-
-			if vp, okp := _locationPreference["zone"]; okp {
-				settings.LocationPreference.Zone = vp.(string)
-			}
-		}
-	}
-
-	if v, ok := _settings["maintenance_window"]; ok {
-		windows := v.([]interface{})
-		if len(windows) > 0 && windows[0] != nil {
-			settings.MaintenanceWindow = &sqladmin.MaintenanceWindow{}
-			window := windows[0].(map[string]interface{})
-
-			if vp, okp := window["day"]; okp {
-				settings.MaintenanceWindow.Day = int64(vp.(int))
-			}
-
-			if vp, okp := window["hour"]; okp {
-				settings.MaintenanceWindow.Hour = int64(vp.(int))
-			}
-
-			if vp, ok := window["update_track"]; ok {
-				if len(vp.(string)) > 0 {
-					settings.MaintenanceWindow.UpdateTrack = vp.(string)
-				}
-			}
-		}
-	}
-
-	if v, ok := _settings["pricing_plan"]; ok {
-		settings.PricingPlan = v.(string)
-	}
-
-	if v, ok := _settings["replication_type"]; ok {
-		settings.ReplicationType = v.(string)
-	}
-
-	if v, ok := _settings["user_labels"]; ok {
-		settings.UserLabels = convertStringMap(v.(map[string]interface{}))
-	}
-
-	instance := &sqladmin.DatabaseInstance{
-		Region:          region,
-		Settings:        settings,
-		DatabaseVersion: databaseVersion,
-	}
 
+	var name string
 	if v, ok := d.GetOk("name"); ok {
-		instance.Name = v.(string)
+		name = v.(string)
 	} else {
-		instance.Name = resource.UniqueId()
-		d.Set("name", instance.Name)
+		name = resource.UniqueId()
 	}
 
-	if v, ok := d.GetOk("replica_configuration"); ok {
-		_replicaConfigurationList := v.([]interface{})
-
-		if len(_replicaConfigurationList) == 1 && _replicaConfigurationList[0] != nil {
-			replicaConfiguration := &sqladmin.ReplicaConfiguration{}
-			mySqlReplicaConfiguration := &sqladmin.MySqlReplicaConfiguration{}
-			_replicaConfiguration := _replicaConfigurationList[0].(map[string]interface{})
-
-			if vp, okp := _replicaConfiguration["failover_target"]; okp {
-				replicaConfiguration.FailoverTarget = vp.(bool)
-			}
-
-			if vp, okp := _replicaConfiguration["ca_certificate"]; okp {
-				mySqlReplicaConfiguration.CaCertificate = vp.(string)
-			}
-
-			if vp, okp := _replicaConfiguration["client_certificate"]; okp {
-				mySqlReplicaConfiguration.ClientCertificate = vp.(string)
-			}
+	d.Set("name", name)
```

Comment:
This is unusual! Do you know why we do this?

Reply:
Originally we had a `d.Set` under the else block, but I pulled it out because this block mirrors a `name_prefix` block such as [in SslCertificate](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_ssl_certificate.go#L319). I don't think the d.Set is necessary at all b/c of the Read, but I figured we can converge on a single shape for automatically generated names.